Enhance ledstrip BEACON profile

etracer65 opened the pull request:
Added features to the BEACON ledstrip profile:

Added a `ledstrip_beacon_color` to allow color selection rather than forcing to white.

Added `ledstrip_beacon_period_ms` to configure the blink period in milliseconds. Smaller time periods mean faster blinking.

Added `ledstrip_beacon_percent` to configure the "ON" time duty cycle. User can set to 100% to have the beacon display a solid color. 0% can be used to turn the becaon completely off.

Added `ledstrip_beacon_armed_only` to allow the user to configure whether the beacon is only on when armed.

Added the new parameters to the CMS menu.

Simplified the code and combined the RACE and BEACON profile processing.

etracer65 commented:
Adds 128 bytes to the simplified `USE_LED_STRIP` on F3.

stawiski commented on src/main/io/ledstrip.c:
```diff
@@ -80,25 +80,23 @@
 
 PG_REGISTER_WITH_RESET_FN(ledStripConfig_t, ledStripConfig, PG_LED_STRIP_CONFIG, 0);
 
+#define COLOR_UNDEFINED 255
```
Isn't a better place for this next to, or even inside the ```colorId_e``` enum?

etracer65 replied:
No. It's not a real color and shouldn't be part of the enum (and would change the count inappropriately), and it's only used in ledstrip.c so there's no need for it to be in the header.

etracer65 commented on src/main/io/ledstrip.c:
```diff
@@ -1265,15 +1274,17 @@ void setLedProfile(uint8_t profile)
-void setLedRaceColor(uint8_t color)
+void setLedBeaconColor(uint8_t color)
 {
     if (color <= COLOR_DEEP_PINK) {
```
You're commenting on code that's not even part of this PR. But yes it would probably be better to use the count.

stawiski replied:
In ```setLedRaceColor()``` that line is a part of your PR :)

etracer65 commented on src/main/io/ledstrip.c:
```diff
@@ -1087,28 +1089,38 @@ static uint8_t selectVisualBeeperColor(uint8_t colorIndex)
+            break;
+
+        case LED_PROFILE_BEACON: {
+            if ((!ledStripConfig()->beaconArmedOnly || ARMING_FLAG(ARMED))
```
I see your point. How about changing to `ledstrip_beacon_mode` with `ALWAYS | AFTER_ARMING | ARMED_ONLY` ? The `AFTER_ARMING` option would only enable after the initial arming but would stay on after that.

Regarding failsafe: Seems like in `RACE` or `BEACON` profiles we could just "force" a special state when in failsafe. When in `STATUS` (full-featured) mode the user can enable warnings which will include the failsafe state, but in `RACE` or `BEACON` there's no way to display this. So I'm thinking force it into a fast beacon with white LED's whenever in failsafe. The idea would be to shoot for maximum visibility.

mikeller replied:
> Regarding failsafe: Seems like in RACE or BEACON profiles we could just "force" a special state when in failsafe. 

I think this is pretty much what the use case for the original 'BEACON' implementation was - let's go for this. Probably throw in 'when in failsafe or RX_SET', to cater for these times when you hit the bushes right next to you.

Not sure if we even need the `AFTER_ARMING` option in this case.

etracer65 commented:
Rebased and pushed changes based on discussion. Added an overriding blinking mode if `RX_SET` or failsafe condition. This overrides any settings that may prevent the indicator from displaying (with the exception of the `LEDLOW` mode which always suppresses the LED output).

Added a `ledstrip_visual_beeper_color` because previously the color was hardcoded to "ORANGE". Turns out the default color for the `RACE` profile was also orange. A flashing LED indicator doesn't show up well against the same color background. 😉 To allow the user to avoid this type of conflict the setting was made configurable.
